js/MatomoOptOutManager.js

```diff
@@ -0,0 +1,230 @@
+			if (useTracker) {
+				content += '<input onclick="_paq.push([\'forgetUserOptOut\']); MatomoOptOutManager.consentGiven();" id="trackVisits" type="checkbox" />';
+			} else {
+				content += '<input onclick="MatomoOptOutManager.consentGiven();" id="trackVisits" type="checkbox" />';
+			}
+			content += '<label for="trackVisits"><strong><span>' + this.settings.YouAreOptedOut + ' ' + this.settings.CheckToOptIn + '</span></strong></label>';
+		}
+		div.innerHTML = content;
+		const hideButton = document.createElement("button");
+		hideButton.textContent = "esconder";
+		hideButton.onclick = () => {
+			const div = document.getElementById(this.settings.divId);
+			div.style.display = "none";
+		};
+		div.appendChild(hideButton);
+	}
+
+	static initMatomoConsentManager(useSecureCookies, cookiePath, cookieDomain, cookieSameSite) {
+		MatomoOptOutManager.useSecureCookies = useSecureCookies;
+		MatomoOptOutManager.cookiePath = cookiePath;
+		MatomoOptOutManager.cookieDomain = cookieDomain;
+		MatomoOptOutManager.cookieSameSite = cookieSameSite;
+		MatomoOptOutManager.CONSENT_COOKIE_NAME = 'mtm_consent';
+		MatomoOptOutManager.CONSENT_REMOVED_COOKIE_NAME = 'mtm_consent_removed';
+		if (useSecureCookies && location.protocol !== 'https:') {
+			console.log('Error with setting useSecureCookies: You cannot use this option on http.');
+		} else {
+			MatomoOptOutManager.cookieIsSecure = useSecureCookies;
+		}
+	}
+
+	static hasConsent() {
+		const consentCookie = MatomoOptOutManager.getCookie(MatomoOptOutManager.CONSENT_COOKIE_NAME);
+		const removedCookie = MatomoOptOutManager.getCookie(MatomoOptOutManager.CONSENT_REMOVED_COOKIE_NAME);
+
+		if (!consentCookie && !removedCookie) {
+
+
+			return true;
+		}
+		if (removedCookie && consentCookie) {
+			MatomoOptOutManager.setCookie(MatomoOptOutManager.CONSENT_COOKIE_NAME, '', -129600000);
+
+			return false;
+		}
+		return (consentCookie || consentCookie !== null);
+	}
+
+
+
+	static getCookie(cookieName) {
+		const cookiePattern = new RegExp('(^|;)[ ]*' + cookieName + '=([^;]*)');
+		const cookieMatch = cookiePattern.exec(document.cookie);
+		return cookieMatch ? window.decodeURIComponent(cookieMatch[2]) : null;
+	}
+
+	static setCookie(cookieName, value, msToExpire) {
+		const expiryDate = new Date();
+		expiryDate.setTime((new Date().getTime()) + msToExpire);
+		document.cookie = cookieName + '=' + window.encodeURIComponent(value) +
+			(msToExpire ? ';expires=' + expiryDate.toGMTString() : '') +
+			';path=' + (MatomoOptOutManager.cookiePath || '/') +
+			(MatomoOptOutManager.cookieDomain ? ';domain=' + MatomoOptOutManager.cookieDomain : '') +
+			(MatomoOptOutManager.cookieIsSecure ? ';secure' : '') +
+			';SameSite=' + MatomoOptOutManager.cookieSameSite;
+		if ((!msToExpire || msToExpire >= 0) && MatomoOptOutManager.getCookie(cookieName) !== String(value)) {
+			console.log('There was an error setting cookie `' + cookieName + '`. Please check domain and path.');
+		}
+	}
+	static consentRevoked() {
```

Comment:
O proprio matomo tem funcionalidade de consentimento, 
`_paq.push(['optUserOut']);` e `_paq.push(['forgetUserOptOut']);`

Ver documentação https://developer.matomo.org/guides/tracking-javascript-guide#asking-for-consent

Aparenta estar a ser implementada uma funcionalidade redundante